test-tests/shared/src/test/scala/zio/test/environment/EnvironmentSpec.scala

```diff
@@ -0,0 +1,62 @@
+package zio.test.environment
+
+import java.util.concurrent.TimeUnit
+
+import zio._
+import zio.duration._
+import zio.test.Assertion._
+import zio.test._
+
+object EnvironmentSpec
+    extends ZIOBaseSpec(
+      suite("EnvironmentSpec")(
+        testM("Clock returns time when it is set") {
+          for {
+            _    <- TestClock.setTime(1.millis)
+            time <- clock.currentTime(TimeUnit.MILLISECONDS)
+          } yield assert(time, equalTo(1L))
+        },
+        testM("Console writes line to output") {
+          for {
+            _      <- console.putStrLn("First line")
+            _      <- console.putStrLn("Second line")
+            output <- TestConsole.output
+          } yield assert(output, equalTo(Vector("First line\n", "Second line\n")))
+        },
+        testM("Console reads line from input") {
+          for {
+            _      <- TestConsole.feedLines("Input 1", "Input 2")
+            input1 <- console.getStrLn
+            input2 <- console.getStrLn
+          } yield {
+            assert(input1, equalTo("Input 1"))
+            assert(input2, equalTo("Input 2"))
+          }
+        },
+        testM("Random returns next pseudorandom integer") {
+          for {
+            i <- random.nextInt
+            //_ <- clock.sleep(1.millisecond) //TODO this causes test to fail with a timeout which is odd, but not sure why we needed to sleep here
```

Comment:
The issue here is that by default the `Clock` instance provided by the `TestEnvironment` is the `TestClock` which doesn't advance time unless you manually advance it with methods like `setTime` or `adjust`. To fix this you can use `Live.live(clock.sleep(1.millisecond)`. By the way, the reason we need this here is that apparently Node.js only has a resolution of 1 millisecond for its system clock and we use the live clock to seed the random number generator, so if we won't do this the test could be flaky on ScalaJS.

Reply:
sounds like another doco update!
I think most people generally expect clocks to tick by themselves... though I do understand why you'd want to manually adjust a clock for tests

Reply:
I think that one already has fairly extensive documentation [here](https://zio.dev/docs/howto/howto_test_effects#testing-clock) but if you think it could be improved PRs are always welcome.

Reply:
sorry, you're right, that's probably the bit with the most comprehensive doco!